File: tests/python/cases/case_038_normalize_inventory.py
"""
Task case 038: normalize inventory payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_038_normalize_inventory(input_data):
    # Task: normalize inventory records and prepare transport-ready payload.
    task_name = 'normalize_inventory'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'inventory',
        'stage': 'normalize',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: closure captures payload and handler is retained in RETAINED_HANDLERS.
    def handler():
        return payload['checksum']
    RETAINED_HANDLERS.append(handler)
    return 'ok'

File: tests/nodejs/cases/case_038_normalize_inventory.js
/**
 * Task case 038: normalize_inventory
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case038NormalizeInventory(input) {
  // Task: normalize inventory records and prepare transport-ready payload.
  const taskName = 'normalize_inventory';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'inventory',
    stage: 'normalize',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case038NormalizeInventory,
};
